mitodl/mit-open
# Search page: paginated, resource_type tabs

### What are the relevant tickets?
Toward #490 

### Description (What does it do?)
This adds a fairly basic search page supporting:
- search text
- tabbed facet for resource types
- pagination (not infinite scroll)

### Screenshots (if appropriate):

This is a **work in progress**; CSS needs work.

<img width="1017" alt="Screenshot 2024-03-01 at 3 24 40 PM" src="https://github.com/mitodl/mit-open/assets/9010790/e396852a-1811-400a-9ad1-f15705e1cac3">

```
                                                                    ⋮
```

<img width="1009" alt="Screenshot 2024-03-01 at 3 24 50 PM" src="https://github.com/mitodl/mit-open/assets/9010790/2dff5076-5d4f-46de-b46c-6de0ea2b8d26">

### How can this be tested?
1. Restart docker (at least watch container) to pick up dependency change
2. Visit http://localhost:8063/search
3. Try some search queries:
    - search text
    - different tabs ("All", "Courses", "Programs", "Podcasts")
    - View different pages of the search
4. Be sure to try forward/backward navigation, too. It should behave as you would expect.

### TODO:
- [ ] Clean up CSS
